core/shared/src/main/scala/zio/ZRef.scala

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio
+import java.util.concurrent.atomic.AtomicReference
+
+/**
+ * A `ZRef[EA, EB, A, B]` is a mutable atomic reference for the `IO` monad. This is the `IO` equivalent of
```

Comment:
I don't think we still use "the `IO` monad" idiom anymore, how about:
```suggestion
 * A `ZRef[EA, EB, A, B]` is a purely functional mutable reference augmented with atomic operations.
```